h2/src/main/org/h2/mode/FunctionsPostgreSQL.java

```diff
@@ -221,6 +231,29 @@ protected Value getValueWithArgs(Session session, Expression[] args) {
             // Not implemented
             result = v1.convertTo(Value.VARCHAR);
             break;
+        case ARRAY_TO_STRING:
+            StringJoiner joiner = new StringJoiner(v1.getString());
+            if (v0.getValueType() != Value.ARRAY) {
+                throw DbException.getInvalidValueException("ARRAY_TO_STRING array", v0);
+            }
+            if (v2 != null) {
+                int v2Type = v2.getValueType();
+                if (v2Type > Value.VARCHAR_IGNORECASE) {
+                    throw DbException.getInvalidValueException("ARRAY_TO_STRING optional_null_string", v2);
+                }
```

Comment:
What do you want from its value type and why this condition accepts many random data types?

Use
```Java
String nullString = null;
if (v2 != null) {
    nullString = v2.getString();
}
```
as it was suggested earlier. Additional checks not worth here.

Reply:
But if you want them for a some reason, use `DataType.isCharacterStringType()` and preserve special case for `NULL`.

Reply:
PostgreSQL also doesn't accept other data types in the second argument, but you don't check it, so I don't understand why you want to check the optional third argument.

First argument is very special and special check for it simplifies implementation, others are simple string arguments, from my point of view these additional conditions for them aren't required. H2 is not designed to validate SQL for other databases anyway.

Reply:
I just forgot to add 2nd arg type check before. Now 3rd arg type check removed, which made code simpler : - )